packages/xo-server/src/xapi/mixins/patching.js

```diff
@@ -347,7 +346,8 @@ export default {
       )
     }
 
-    return asyncMap(hosts, async host => {
+    hosts = hosts.sort(({ $ref }) => ($ref === this.pool.master ? -1 : 1))
```

Comment:
Maybe a small comment explaining why this is necessary?

Reply:
Simply something like: “patches must be installed sequentially, host by host, with pool masters first”?